### DISCUSSION TITLE
Just added my first action and getting a issue with missing namespace

### DISCUSSION THREAD

**Discussion by oliwerhelsen** [lunarstorm/laravel-ddd#62] [Q&A] [answered]
Hi,

I just generated my first action into my project. But I got `Trait "Lorisleiva\Actions\Concerns\AsAction" not found `.
Any idea? Have been following the guide and read the documentation without any luck :(

**Comment by jaspertey** [comment, marked as answer]
> Hi,
> 
> I just generated my first action into my project. But I got `Trait "Lorisleiva\Actions\Concerns\AsAction" not found `. Any idea? Have been following the guide and read the documentation without any luck :(

Hi Oliwer,

By default, the package generates actions based on the `lorisleiva/laravel-actions` package, which means you need to have that package installed: https://github.com/lorisleiva/laravel-actions - if you don't want to use that package and roll your own, then you'd have to publish the stubs and customize `stubs/action.php.stub` accordingly.

Same applies to data transfer objects, which rely on the `spatie/laravel-data` peer dependency unless you customize the `ddd.base_dto` otherwise.

I ought to make the documentation clearer on this, as you'd have to read the comment blocks inside the config section of the readme to notice these comments. I will make a note to update the readme.

**Reply by oliwerhelsen** [reply]
Thanks for the replay @JasperTey. Worked perfectly :-)
